### PR TITLE
harfbuzz: add CoreText support for darwin

### PR DESCRIPTION
**Pull request by LnL7**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthewbauer** [comment]
I dislike adding meta attribute names to all-packages.nix. Could we just have Harfbuzz default to use CoreText with a config option? Users of Harfbuzz shouldn't have to know what it's been using on the backend IIRC.

**Comment by LnL7** [comment]
I'll remove it for now, don't want to enable it by default since it depends on frameworks.

**Comment by matthewbauer** [comment]
I wonder if we can get rid of harfbuzz-icu as well? It looks like we didn't want to enable it by default but a few packages might actually need it (IMO the override should go in the callPackage of where it's needed).

**Comment by LnL7** [comment]
What about harfbuzz and harfbuzzFull, that's pretty common for packages with feature flags. This avoids extra dependencies by default while still making it available. (eg. nix-index will be able to find the `hb-coretext.h` header)

File: pkgs/development/libraries/harfbuzz/default.nix [hunk 1]
```diff
@@ -1,5 +1,7 @@
 { stdenv, fetchurl, pkgconfig, glib, freetype, cairo, libintl
 , icu, graphite2, harfbuzz # The icu variant uses and propagates the non-icu one.
+, ApplicationServices, CoreText
+, withCoreText ? false
```
Comment (matthewbauer) [inline pkgs/development/libraries/harfbuzz/default.nix]: Also see `withApple' in #39580.

**Comment by xeji** [comment]
This looks ready but there's a merge conflict now.

**Pull request merged**